[docs] eas build: update docs on the credentialsSource option (#12616)
expo/expo: docs/pages/build/eas-json.md

- `"workflow": "generic"` indicates that your project is a generic one.
- `extends` specifies name of build profile that current profile inherits values from.

> - `credentialsSource` defines the source of credentials for this build profile. If you want to provide your own `credentials.json` file, set this to `local` ([learn more on this here](/app-signing/local-credentials.md)). If you want to use the credentials EAS already has stored for you, choose `remote` (this is the default option).

Reviewer: ```suggestion
- `credentialsSource` defines the source of credentials for this build profile. If you want to provide your own `credentials.json` file, set this to `local` ([learn more on this here](/app-signing/local-credentials.md)). If you want to use the credentials managed by EAS, choose `remote` (this is the default option).
```
